kernels/haskell.nix

```diff
@@ -1,41 +1,41 @@
-{
-  name,
-  availableKernels,
-  extraArgs,
-}:
-availableKernels.haskell {
-  inherit (extraArgs) system;
-  name = "custom-${name}"; # must be unique
-  displayName = "custom ${name}";
-  extraHaskellPackages = p: [
-    p.hvega
-    p.lens
-    p.log-domain
-    p.katip
-    p.ihaskell-hvega
-    p.ihaskell-diagrams
-    p.text
-    p.diagrams
-    p.diagrams-cairo
-    p.aeson
-    p.lens
-    p.lens-aeson
-    p.pretty-simple
-    p.monad-loops
-    p.hamilton
-    p.hmatrix
-    p.vector-sized
-    p.linear
-    p.recursion-schemes
-    p.data-fix
-    p.free
-    p.comonad
-    p.adjunctions
-    p.distributive
-    p.vector
-    p.megaparsec
-    p.histogram-fill
-    p.gloss
-    extraArgs.monad-bayes
-  ];
+{monad-bayes}:
+
+{pkgs, ...}: {
+  kernel.haskell.monad-bayes = {
+    enable = true;
+    name = "monad-bayes";
+    displayName = "monad-bayes";
+    extraHaskellPackages = p: [
+      p.hvega
+      p.lens
+      p.log-domain
+      p.katip
+      p.ihaskell-hvega
+      p.ihaskell-diagrams
+      p.diagrams
+      p.diagrams-cairo
+      p.aeson
+      p.lens
+      p.lens-aeson
+      p.pretty-simple
+      p.monad-loops
+      # hamilton seems unmaintained, unclear whether we can keep it. See https://github.com/tweag/monad-bayes/issues/300
+      # p.hamilton
+      p.hmatrix
+      p.vector-sized
+      p.linear
+      p.recursion-schemes
+      p.data-fix
+      p.free
+      p.comonad
+      p.adjunctions
+      p.distributive
+      p.vector
+      p.megaparsec
+      p.histogram-fill
+      # Strange build error
```

Comment:
I should try and fix this

Reply:
I couldn't. I tried overriding in any way I could think of, but it seems like `gloss` in `ihaskell` is broken because they pin `bytestring` to the wrong version.

Reply:
The only way I see for now is leaving this out, then one of the notebooks will not work out of the box.